protobuf/protoc-gen-govalidator/plugin/plugin.go

```diff
@@ -242,6 +242,16 @@ func (p *plugin) generateLengthValidator(variableName string, ccTypeName string,
 
 func (p *plugin) generateInnerMessageValidator(variableName string, ccTypeName string, fieldName string, field *descriptor.FieldDescriptorProto, nullable bool) {
 
+	if getMsgRequiredValue(field) {
+		if nullable {
+			p.P(`if `, variableName, ` == nil {`)
+			p.In()
+			p.P(`return fmt.Errorf("message `, fieldName, ` is required")`)
+			p.Out()
+			p.P(`}`)
+		}
+	}
+
 	if nullable {
 		p.P(`if `, variableName, ` != nil {`)
 		p.In()
```

Comment:
```suggestion
		p.In()
		if getMsgRequiredValue(field) {
		    p.P(`return fmt.Errorf("message `, fieldName, ` is required")`)
		    p.Out()
		    p.P(`}`)
		}	
```

Reply:
This is already past the `!= nil` check. This won't output the correct code.